twitchio/ext/pubsub/models.py

```diff
@@ -339,7 +342,77 @@ def __init__(self, client: Client, topic: str, data: dict):
         if data["message"]["data"]["updated_at"]:
             self.updated_at = parse_timestamp(data["message"]["data"]["expires_at"])
 
+class PubSubChannelSubscribe(PubSubMessage):
+    """
+    Channel subscription
 
+    Attributes
+    -----------
+    channel: :class:`twitchio.Channel`
+        Channel that has been subscribed or subgifted.
+    channel_id: :class:`int`
+        The channel id the action occurred on.
+    context: :class:`str`
+        Event type associated with the subscription product.
+    user: :class:`twitchio.PartialUser`
+        The person who subscribed or sent a gift subscription.
+    message: :class:`str`
+        Message sent with the sub/resub
+    recipient: :class:`twitchio.PartialUser`
+        The person the who received the gift subscription.
+    sub_plan: :class:`str`
+        Subscription Plan ID.
+    sub_plan_name: :class:`str`
+        Channel Specific Subscription Plan Name.
+    time: :class:`datetime.datetime`
+        Time when the subscription or gift was completed. RFC 3339 format.
+    cumulative_months: :class:`str`
+        Cumulative number of tenure months of the subscription.
+    streak_months: :class:`str`
+        Denotes the user's most recent (and contiguous) subscription tenure streak in the channel.
+    is_gift: :class:`bool`
+        If this sub message was caused by a gift subscription,
+    multi_month_duration: :class:`int`
+        Number of months gifted as part of a single, multi-month gift OR number of months purchased as part of a multi-month subscription.
+    """
+
+    __slots__ = ("channel",
+        "channel_id",
+        "context",
+        "user",
+        "message",
+        "is_gift",
+        "recipient",
+        "sub_plan",
+        "sub_plan_name",
+        "time",
+        "cumulative_months",
+        "streak_months",
+        "multi_month_duration",)
+
+
+    def __init__(self, client: Client, topic: str, data: dict):
+        super().__init__(client, topic, data)
+
+        subscription = data["message"]
+        
+        self.channel: Channel = client.get_channel(subscription["channel_name"]) or Channel(
+            name=subscription["channel_name"], websocket=client._connection
+        )
+        self.channel_id: int = int(subscription["channel_id"])
+        self.context: str = subscription["context"]
+        self.user = PartialUser(client._http, subscription["user_id"], subscription["user_name"], subscription["display_name"]) if self.context != "anongiftsub" or self.context != "anonresubgift" else None
```

Comment:
It would probably be better to check if self.context not in set { } 
Also PartialUser does not take display_name